docs/pages/pmd/devdocs/major_contributions/adding_new_language.md

```diff
@@ -27,9 +27,9 @@ folder: pmd/devdocs
 *   Note: These AST node classes are generated usually once by javacc/jjtree and can then be modified as needed.
 
 ## 4.  Compile your parser (if using JJT)
-*   An ant script is being used to compile jjt files into classes. This is in alljavacc.xml file.
-*   In the file, create a new target for your language. Use vmjjtree or javajjtree as an example.
-*   Inside the alljavacctarget, add your new target to the “depends” list just before cleanup
+*   An ant script is being used to compile jjt files into classes. This is in `pmd-<lang>/src/main/ant/alljavacc.xml` file.
+*   Create `alljavacc.xml` file for your language, you can use one from `pmd-java` as an example.
+*   To generate stubs for AST nodes of your language, temporary `<!-- comment out -->` the `<delete>` tag near the bottom of the file, then copy generated files from `pmd-<lang>/target/generated-sources` (you need files starting with `AST` only).
```

Comment:
Note, that deleting all generated AST nodes is done for pmd-java. Other languages (like pmd-plsql) selectively delete only some nodes. In general, you want to copy the generated AST nodes, because you want to modify them and add more useful methods (like getters for specific information).

Reply:
Thanks for clarification! I will fix it slightly later, when I would try it myself as part of #2041.